lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2018, Ian Tewksbury <itewk@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: manageiq_config
+
+short_description: Module for managing ManageIQ configuration.
+version_added: '2.5'
+author: Ian Tewksbury (@itewk)
+description:
+    - "Module for managing ManageIQ or CloudForms Managment Engine (CFME) configuration via the local rails runner on the Appliance."
+
+options:
+    name:
+        description:
+            - Name of the ManageIQ config element to modify.
+        required: True
+    value:
+        description: Dictionary to set as the value for the ManageIQ config option. Any values not given will be left at current value.
+        required: False
+        default: {}
+
+'''
+
+EXAMPLES = '''
+# set the smtp settings
+- manageiq_config:
+    name: smtp
+    value:
+      from: cfme@example.com
+      host: postfix.example.com
+      port: 25
+      domain: example.com
+
+# set generic worker memory threshold
+- manageiq_config:
+    name: workers
+    value:
+      worker_base:
+        queue_worker_base:
+          generic_worker:
+            memory_threshold: '900.megabytes'
+'''
+
+RETURN = '''
+name:
+    description: Name of the ManageIQ config option that is being modified.
+    type: str
+    returned: always
+original_value:
+    description: The origional value of the ManageIQ config option being modified before modification.
+    type: dict
+    returned: always
+value:
+    description: The value of the ManageIQ config option being modified after modification.
+    type: dict
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import json
+import time
+import collections
+import copy
+
+VMDB_PATH = '/var/www/miq/vmdb'
+CONFIRM_UPDATE_MAX_RETRY = 10
```

Comment:
This should be configurable in the module arguments.

Reply:
okay.